casepro/cases/tasks.py

```diff
@@ -13,3 +14,40 @@ def case_export(export_id):
     logger.info("Starting case export #%d..." % export_id)
 
     CaseExport.objects.get(pk=export_id).do_export()
+
+
+@shared_task(ignore_result=True)
+def get_all_cases_passed_response_time():
+    from .models import Case
+    cases = Case.get_all_passed_response_time()
+    tasks = []
+    for case in cases:
+        tasks.append(reassign_case.s(case.pk))
+
+    if tasks:
+        group_task = group(tasks)
+        group_task.apply_async()
+
+
+@shared_task(ignore_result=True)
+def reassign_case(case_id):
+    from .models import Case, CaseAction, SystemUser
+    try:
+        case = Case.objects.get(pk=case_id)
+    except Case.DoesNotExist:
+        logger.warn("Could not load case #%d for re-assignment" % case_id)
+        return
+
+    if not case.has_passed_response_time:
+        # the case has been reassigned more recently that our expected window, do nothing
+        return
+
+    action = case.actions.filter(action=CaseAction.REASSIGN).latest('created_on')
+    if action.created_by is None:
```

Comment:
to confirm: Yes, the `created_by` field will contain the instance of `SystemUser` if it was reassigned by the system.